### ISSUE TITLE
gdbus_codegen should return two different targets (.c, .h)

### ISSUE THREAD

**Issue by Salamandar** [mesonbuild/meson#3655]
With glib<2.56.2, gdbus_codegen() returns twice the same custom_target, with two outputs.

A bug occurs when this return value is used for generate_gir(). The source list then contains twice the .c, and zero times the .h…

Maybe gdbus_codegen() should return two custom_targets, with each one output (once the .c, once the .h).

EDIT : Actually, after reading the source of generate_gir, I suppose the bug is there.

```
                    gir_filelist.write(os.path.join(state.environment.get_build_dir(),
                                                    state.backend.get_target_dir(s),
                                                    s.get_outputs()[0]) + '\n')
```
Why only `s.get_outputs()[0]` ?

**Comment by nirbheek** [comment]
Yeah, that `s.get_outputs()[0]` is probably wrong. It should loop over all the outputs and write them all.

**Comment by tintou** [comment]
This issue is composed of two issues solved in PR #3744 for looping over the outputs and #3743 for the misbehavior of gdbus_codegen
